### PR TITLE
Update urls to use https instead of http

### LINKED ISSUE #1333: Change Http to Https
### Bug Report
Due to App Transport Security on iOS special exceptions are needed to connect to HTTP, so the iOS team will update urls to use HTTPS. Although Android can allow HTTP by setting    "android:usesCleartextTraffic="true"" in the Android manifest we want to ensure we are aligned with iOS. Therefore The java suite should update urls to use HTTPS instead of HTTP

### PR DESCRIPTION
**Pull request by RHenigan**
Fixes #1333 

This PR is **[ready]** for review.

### Risk
This PR makes **[no]** API changes.

### Summary
replace http with https where needed

### Test Steps
- Add logging to SDLProxyBase and PolicyFetcher to observe URLs
- Run Android Test App and connect to manticore
- Update the policy table url in manticore
- Observe Logs to ensure Http is being replaced with Https in URLs and the app is able to communicate with the policy table

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/LockScreenManager.java [hunk 1]
```diff
@@ -232,6 +232,7 @@ public void onNotified(RPCNotification notification) {
 							msg.getUrl() != null) {
 						// send intent to activity to download icon from core
 						deviceIconUrl = msg.getUrl();
+						deviceIconUrl = deviceIconUrl.replaceFirst("http://", "https://");
```
Comment (bilal-alsharifi) [inline android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/LockScreenManager.java:235]: I see this is done in the icon caching PR so there is no need to make this change here

File: base/src/main/java/com/smartdevicelink/proxy/rpc/OnSystemRequest.java [hunk 2]
```diff
@@ -1,34 +1,34 @@
...
- * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
- * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
- * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
- * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
- * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
- * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
- * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
- * POSSIBILITY OF SUCH DAMAGE.
- */
+/*
```
Comment (bilal-alsharifi) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/OnSystemRequest.java:1]: Can we discard this change?

File: android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/LockScreenManager.java (hunk 1 above)
Comment (RHenigan) [reply android/sdl_android/src/main/java/com/smartdevicelink/managers/lockscreen/LockScreenManager.java]: Merged develop in and took the changes from the lockscreen caching

File: base/src/main/java/com/smartdevicelink/proxy/rpc/OnSystemRequest.java (hunk 2 above)
Comment (RHenigan) [reply base/src/main/java/com/smartdevicelink/proxy/rpc/OnSystemRequest.java]: the only changes in this file were the comments done by intellij so I checked out the file from develop

File: android/sdl_android/src/main/java/com/smartdevicelink/proxy/SdlProxyBase.java [hunk 3]
```diff
@@ -4126,7 +4127,7 @@ public void run() {
 						onRPCNotificationReceived(msg);
 					}
 				} else {
-					updateBroadcastIntent(sendIntent, "COMMENT1", "Sending to cloud: " + msg.getUrl());
+					updateBroadcastIntent(sendIntent, "COMMENT1", "Sending to cloud: " + msg.getUrl().replaceFirst("http://", "https://"));
```
Comment (bilal-alsharifi) [inline android/sdl_android/src/main/java/com/smartdevicelink/proxy/SdlProxyBase.java:4130]: It is not necessary to update that one

File: android/sdl_android/src/main/java/com/smartdevicelink/proxy/SdlProxyBase.java [hunk 4]
```diff
@@ -4303,10 +4304,10 @@ public void run() {
 						Thread handleOffBoardTransmissionThread = new Thread() {
 							@Override
 							public void run() {
-								byte[] file = FileUtls.downloadFile(msg.getUrl());
+								byte[] file = FileUtls.downloadFile(msg.getUrl().replaceFirst("http://", "https://"));
```
[the remark this answers is not in the document]
Comment (RHenigan) [reply android/sdl_android/src/main/java/com/smartdevicelink/proxy/SdlProxyBase.java]: resolved

File: base/src/main/java/com/smartdevicelink/managers/lifecycle/PoliciesFetcher.java [hunk 5]
```diff
@@ -92,7 +92,7 @@ private static HttpURLConnection getURLConnection(Headers myHeader, String sURLS
         }
 
         try {
-            url = new URL(sURLString);
+            url = new URL(sURLString.replaceFirst("http://", "https://"));
             urlConnection = (HttpURLConnection) url.openConnection();
             urlConnection.setConnectTimeout(CONNECTION_TIMEOUT);
             urlConnection.setDoOutput(bDoOutput);
```
[the remark this answers is not in the document]
Comment (RHenigan) [reply base/src/main/java/com/smartdevicelink/managers/lifecycle/PoliciesFetcher.java:98]: added these and looking through to make sure others are not missed

File: android/sdl_android/src/main/java/com/smartdevicelink/proxy/SdlProxyBase.java (hunk 3 above)
Comment (RHenigan) [reply android/sdl_android/src/main/java/com/smartdevicelink/proxy/SdlProxyBase.java]: removed

File: base/src/main/java/com/smartdevicelink/managers/lifecycle/PoliciesFetcher.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (RHenigan) [reply base/src/main/java/com/smartdevicelink/managers/lifecycle/PoliciesFetcher.java:98]: updated

**Pull request merged**